### ISSUE TITLE
RUF106's fix un-suppresses previously suppressed diagnostics

### ISSUE THREAD

**Issue by kdebrab** [astral-sh/ruff#28007]
### Summary

# RUF106's fix un-suppresses previously suppressed diagnostics

## Summary

The preview RUF106 fix can drop rule codes out of `# ruff: ignore[...]` comments when activated in combination with I001 and RUF100.

## Reproduction

`ruff.toml`:

```toml
line-length = 88
preview = true

[lint]
select = ["I001", "PLC0415", "PLC2701", "RUF100", "RUF106"]
```

`case_a.py` — a suppression comment carrying two codes:

```python
def f() -> None:
    from some.pkg.mod import _private_thing  # ruff: ignore[PLC0415, PLC2701]

    print(_private_thing)
```

`case_b.py` — a suppression comment carrying one code:

```python
from some.external.pkg import _private_a, _private_b, public_c  # ruff: ignore[PLC2701]

print(_private_a, _private_b, public_c)
```

Then:

```console
$ ruff check --fix --output-format=concise .
case_a.py:2:5: import-outside-top-level: `import` should be at the top-level of a file
case_b.py:2:5: import-private-name: Private name import `_private_a` from external module `some.external.pkg`
case_b.py:3:5: import-private-name: Private name import `_private_b` from external module `some.external.pkg`
Found 10 errors (7 fixed, 3 remaining).
```

### Actual output

`case_a.py` — `PLC0415` / `import-outside-top-level` has been dropped from the comment:

```python
def f() -> None:
    from some.pkg.mod import (
        _private_thing,  # ruff: ignore[import-private-name]
    )

    print(_private_thing)
```

`case_b.py` — the suppression comment is gone entirely:

```python
from some.external.pkg import (
    _private_a,
    _private_b,
    public_c,
)

print(_private_a, _private_b, public_c)
```

### Expected output

Conversion of rule codes to human-readable rule names without introducing any lint errors. For `case_a.py` that would be e.g.

```python
def f() -> None:
    # ruff: ignore[import-outside-top-level, import-private-name]
    from some.pkg.mod import _private_thing

    print(_private_thing)
```

## Root cause (according to Opus 5)

A three-rule fix cascade within a single `ruff check --fix` run:

1. **`RUF106`** rewrites `[PLC0415, PLC2701]` to `[import-outside-top-level, import-private-name]`.
   Names are much longer than codes (+29 characters here), so the line now exceeds `line-length`.
2. **`I001`** (isort) reacts to the over-long line by wrapping the import into the parenthesised
   form. In doing so it **relocates the trailing `# ruff: ignore[...]` comment to a different
   line** than the one the suppressed diagnostics are reported on:
   * `case_a.py`: the comment moves onto the `_private_thing,` member line, but `PLC0415`
     (`import-outside-top-level`) is reported on the `from ... import (` statement line.
   * `case_b.py`: the comment stays on the `from ... import (` line, but `PLC2701`
     (`import-private-name`) is now reported on the `_private_a,` / `_private_b,` member lines.
3. **`RUF100`** then sees codes in the comment that no longer match anything on their line, calls
   them unused, and deletes them — the last code's removal takes the whole comment with it.

So the defect here is:

* **`I001`'s wrap does not preserve suppression-comment semantics.** Moving a trailing
  `# ruff: ignore[...]` between the statement line and a member line changes which diagnostics it
  suppresses. This is pre-existing, but `RUF106` is what makes it fire on previously stable code.

I think a shorter term work-around may be making the I001 fix unsafe for lines that end with a pragma comment?

### Version

ruff 0.16.4

**Comment by kdebrab** [comment]
This looks very similar to #20042 (just replace RUF106 for PLC0415), and thus probably the same underlying defect as #12179.

I hit this when upgrading ruff (from 0.15.21 to 0.16.4) in a private repo with `select = ["ALL"]` and `preview = True`, introducing 9 errors across 5 files, all of this shape (6 lost `import-outside-top-level` suppressions and 3 lost `import-private-name` ones). I suspect the underlying issue may be more frequent with RUF106 given that this rule always lengthens lines, sometimes quite dramatically.

**Comment by ntBre** [comment]
Thanks for the report and good find with #20042. This is also tracked in https://github.com/astral-sh/ruff/issues/27059 and will hopefully be helped soon by https://github.com/astral-sh/ruff/pull/27313. I'll close this in favor of #27059 as the closest duplicate.
